sdk/ai/azure-ai-projects/samples/agents/sample_agents_vector_store_batch_file_search.py

```diff
@@ -11,43 +12,52 @@
     the Azure Agents service using a synchronous client.
 
 USAGE:
-    python sample_agents_vector_store_batch_file_search_async.py
+    python sample_agents_vector_store_batch_file_search.py
 
     Before running the sample:
 
-    pip install azure.ai.projects azure-identity
+    pip install azure.ai.projects azure-identity azure-ai-ml
 
     Set this environment variables with your own values:
     PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
 """
 
 import os
+from azure.ai.ml import MLClient
+from azure.ai.ml.constants import AssetTypes
+from azure.ai.ml.entities import Data
+
 from azure.ai.projects import AIProjectClient
-from azure.ai.projects.models import FileSearchTool, FilePurpose
+from azure.ai.projects.models import FileSearchTool
 from azure.identity import DefaultAzureCredential
 
 
 # Create an Azure AI Client from a connection string, copied from your AI Studio project.
 # At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
 # Customer needs to login to Azure subscription via Azure CLI and set the environment variables
 
+credential = DefaultAzureCredential()
 project_client = AIProjectClient.from_connection_string(
-    credential=DefaultAzureCredential(), conn_str=os.environ["PROJECT_CONNECTION_STRING"]
+    credential=credential, conn_str=os.environ["PROJECT_CONNECTION_STRING"]
 )
 
 with project_client:
 
-    # upload a file and wait for it to be processed
-    file = project_client.agents.upload_file_and_poll(file_path="product_info_1.md", purpose=FilePurpose.AGENTS)
-    print(f"Uploaded file, file ID: {file.id}")
+    azure_client = MLClient.from_config(credential)
+    # We will upload the local file to Azure and will use if for vector store creation.
+    local_data = Data(name="products", path="./product_info_1.md", type=AssetTypes.URI_FILE)
+    # The new data object will contain the Azure ID of an uploaded file,
+    # which is the uri starting from azureml://.
+    data_ul = azure_client.data.create_or_update(local_data)
 
     # create a vector store with no file and wait for it to be processed
+    ds = VectorStorageDataSource(storage_uri=data_ul.path, asset_type="uri_asset")
     vector_store = project_client.agents.create_vector_store_and_poll(file_ids=[], name="sample_vector_store")
     print(f"Created vector store, vector store ID: {vector_store.id}")
 
     # add the file to the vector store or you can supply file ids in the vector store creation
     vector_store_file_batch = project_client.agents.create_vector_store_file_batch_and_poll(
-        vector_store_id=vector_store.id, file_ids=[file.id]
+        vector_store_id=vector_store.id, data_sources=[ds]
     )
     print(f"Created vector store file batch, vector store file batch ID: {vector_store_file_batch.id}")
```

Comment:
We shall have datasources also available in CodeInterpreterToolResource

class CodeInterpreterToolResource(_model_base.Model):
    """A set of resources that are used by the ``code_interpreter`` tool.

    :ivar file_ids: A list of file IDs made available to the ``code_interpreter`` tool. There can
     be a maximum of 20 files
     associated with the tool.
    :vartype file_ids: list[str]
    """

    file_ids: Optional[List[str]] = rest_field()
    """A list of file IDs made available to the ``code_interpreter`` tool. There can be a maximum of
     20 files
     associated with the tool."""

Please add sample for this also